lib/ngx/ssl.lua

```diff
@@ -164,6 +167,25 @@ function _M.server_name()
     return nil, ffi_str(errmsg[0])
 end
 
+function _M.client_addr()
+    local r = getfenv(0).__ngx_req
+    if not r then
+        return error("no request found")
+    end
+
+    local sizep = get_size_ptr()
+
+    local rc = C.ngx_http_lua_ffi_ssl_client_addr(r, charpp, sizep, errmsg)
+    if rc == FFI_OK then
+        return ffi_str(charpp[0], sizep[0])
+    end
+
+    if rc == FFI_DECLINED then
```

Comment:
The called C function doesn't return NGX_DECLINED, maybe we don't need to handle it here.